compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -300,6 +300,11 @@ extends NotFoundMsg(MissingIdentID) {
 class TypeMismatch(val found: Type, expected: Type, val inTree: Option[untpd.Tree], addenda: => String*)(using Context)
   extends TypeMismatchMsg(found, expected)(TypeMismatchID):
 
+  private val shouldSuggestNN =
+    if expected.isValueType then
```

Comment:
Let's add a condition `ctx.mode.is(SafeNulls)` so we only compute the subtyping in explicit nulls